modules/python/src2/gen2.py

```diff
@@ -182,6 +205,93 @@
     }
 """)
 
+stub_header = """import numpy as np
+
+from typing import (
+    Any,
+    Dict,
```

Comment:
dont think you need to import Dict 

"use built-in generics (list, dict, tuple, set), instead of importing them from typing, except in type aliases, in base classes, and for arbitrary length tuples (Tuple[int, ...]);"

https://github.com/python/typeshed/blob/master/CONTRIBUTING.md

Reply:
This. Same thing for `Union` and `Optional`

Reply:
This patch targeting Python 3.6+ since it is default on many platforms in production.

Reply:
> This patch targeting Python 3.6+ since it is default on many platforms in production.

Stub files are not restricted to a python version, since they are never executed. You should always use the most modern syntax supported by type-checkers. That's also why `from __future__ import annotations` is a useless statement in stubs.

Type-checkers also don't actively support EOL versions.

That being said, mypy *does* have some bugs left with pipe unions when checking against python 3.7 .